### PR TITLE
Complete site

### PR DESCRIPTION
**Pull request by Cerdtruir**
Full project for JavaScript to do list

view branch: https://luminous-mandazi-6facd5.netlify.app/

### CODE REVIEW

**Comment by Cerdtruir** [comment]
I need to fix the linter error

**Comment by Cerdtruir** [comment]
Screenshot and links to readme

**Comment by Cerdtruir** [comment]
Don't use innerHTML on user inputted data

**Comment by Cerdtruir** [comment]
Change netlify build once merged

**Comment by Cerdtruir** [comment]
@ikraamg I made the requested changes from the previous pr, let me know if anything else needs changing

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+if (localStorage.getItem('storedTasks')) {
+  todoItems = JSON.parse(localStorage.getItem('storedTasks'));
+  tasksHTML();
+}
+
+const removeComplete = document.body.querySelector('.last-item');
+
+removeComplete.onclick = () => {
+  todoItems = removeCompleted(todoItems, tasksHTML);
```
Comment (ikraamg) [inline src/index.js]: This function only takes one argument right?

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  editTask,
+  removeCompleted,
+} from './tasksArray.js';
+
+let todoItems = [];
+
+const lastItem = document.body.querySelector('.last-item');
+const listMain = document.body.querySelector('.todo-list');
+
+function tasksHTML() {
```
Comment (ikraamg) [inline src/index.js]: I would name this something like "renderHTML". That is what is it doing, right? Functions always do something, so they need to be verbs.

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  todoItems.forEach((task, i) => {
+    task.index = i;
+    i += 1;
+  });
```
Comment (ikraamg) [inline src/index.js:21]: Can be extracted for clarity to a method called:

addIndexAttributeToTasks(todoItems)

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  const stringifiedTasks = JSON.stringify(todoItems);
+  localStorage.setItem('storedTasks', stringifiedTasks);
+  document.body.querySelectorAll('.list-task').forEach((element) => {
+    element.remove();
+  });
```
Comment (ikraamg) [inline src/index.js:27]: Can be extracted for clarity to a method called:

clearAllTasks()

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+    content.append(checkbox, taskText, listMove);
+
+    if (task.completed === true) {
+      content.classList.add('mark-completed');
+      checkbox.innerText = '\u2714';
+    }
+    listMain.insertBefore(content, lastItem);
+    i += 1;
+  });
```
Comment (ikraamg) [inline src/index.js]: Can be extracted for clarity to a method called:

renderTasks(todoItems)

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  document.body.querySelectorAll('.task-description').forEach((element) => {
+    element.contentEditable = true;
+    editTask(element, todoItems, tasksHTML);
+  });
```
Comment (ikraamg) [inline src/index.js]: Can be extracted for clarity to a method called:

makeTasksEditable()

File: src/tasksArray.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    // Save on click away
+    element.addEventListener('blur', () => {
+      if (deletedID === false) {
+        const changedID = this.parentElement.id;
+        array[changedID].description = element.innerHTML;
+      }
+      tasksHTML();
+    });
+  });
+}
```
Comment (ikraamg) [inline src/tasksArray.js]: This function is doing a lot, I think you can split it into two or 3 smaller functions but still export only `editTask`.

**Review by ikraamg — APPROVED** [review]
Very well done 👏 

I left suggestions to improve the quality of the code but nicely done overall with good spec coverage 👌

File: src/index.js (hunk 1 above)
Comment (Cerdtruir) [reply src/index.js]: yeah, i think it was old code or something

File: src/index.js (hunk 2 above)
Comment (Cerdtruir) [reply src/index.js]: Fixed this

File: src/index.js (hunk 3 above)
Comment (Cerdtruir) [reply src/index.js:21]: Fixed this

File: src/index.js (hunk 4 above)
Comment (Cerdtruir) [reply src/index.js:27]: Fixed this

File: src/index.js (hunk 5 above)
Comment (Cerdtruir) [reply src/index.js]: Fixed this

**Pull request merged**